presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -240,6 +240,9 @@ public HttpRemoteTask(Session session,
                 if (state.isDone()) {
                     cleanUpTask();
                 }
+                else {
+                    partitionedSplitCountTracker.setPartitionedSplitCount(getPartitionedSplitCount());
```

Comment:
Not sure I understand the question, but I'll take a shot.  `updateTaskInfo` is called when the scheduler adds work on the task.  This is a callback when we get a status update from the task.  Most importantly, this lets us know that work has finished.